include/seastar/util/assert.hh

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2025 ScyllaDB.
+ */
+
+#include <cstdio>
+
+#pragma once
+
+/// Like assert(), but independent of NDEBUG. Active in all build modes.
+#define SEASTAR_ASSERT(x)                                                      \
+  do {                                                                         \
+    if (!(x)) [[unlikely]] {                                                   \
+      printf("%s:%u: %s: Assertion `%s` failed.\n", __FILE__, __LINE__,        \
+             __PRETTY_FUNCTION__, #x);                                         \
+      __builtin_trap();                                                        \
+    }                                                                          \
+  } while (0)
```

Comment:
4-space indents.

Please call a [[noreturn]] function (you can add it in reactor.cc, no need for a new source file) so that we don't have to bring in \<cstdio> for everyone.

Reply:
Done. I added in `log.cc` to keep it in the `util` directory.